clang/include/clang/CIR/Dialect/IR/CIRDialect.td

```diff
@@ -0,0 +1,46 @@
+//===- CIRTypes.td - CIR dialect types ---------------------*- tablegen -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the CIR dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_CIR_DIALECT_CIR
+#define MLIR_CIR_DIALECT_CIR
+
+include "mlir/IR/OpBase.td"
+
+def CIR_Dialect : Dialect {
+  let name = "cir";
+
+  // A short one-line summary of our dialect.
+  let summary = "A high-level dialect for analyzing and optimizing Clang "
+                "supported languages";
+
+  let cppNamespace = "::mlir::cir";
+
+  let useDefaultAttributePrinterParser = 0;
+  let useDefaultTypePrinterParser = 0;
```

Comment:
Should these follow the usual style guide for naming of member variables (start with a capital letter)?

Reply:
We are following MLIR style guide here and these tablegen ones need to be like this in order for certain c++ code to be generated.

This comment reminds me of something we need to discuss for the rest of CIR: MLIR and clang differs on the coding style, which means that CIRGen code has variables starting with capital letter calling into code using lowecase. So far we've been using a mix: in the dialect source we keep MLIR style, and in CIRGen we keep the clang style. 

We don't have a strong opinion here, we rather apply whatever the community/maintainers care and move on with upstreaming. Thoughts? (cc: @joker-eph)

Reply:
If the comment is on the TableGen code: you don't have a choice here because you're implement the ODS API. These aren't just "data member", they are actually "magic" names used by the tablegen backend.

For the C++ code a few lines below, function like `printType()` are override of virtual methods defined in MLIR.

Reply:
Okay. so this is another case of MLIR not following the usual community policies, so this is outside of the scope of these patches and that's fine by me. Thanks for the explanation!

Reply:
Yes, in the early months of MLIR, there was a big discussion on llvm-dev@ about changing the naming convention to start with lower-cases. And it seemed to us that it converged and so on our side we anticipated and updated all the codebase to use the "new" convention. However this didn't carry forward, I don't remember why...
(The plan was even documented: https://llvm.org/docs/Proposals/VariableNames.html )

Reply:
Thread: https://groups.google.com/g/llvm-dev/c/i7TLvzd1OAw/m/cJg8sGnMDAAJ